neqo-transport/src/qlog.rs

```diff
@@ -0,0 +1,314 @@
+            offset.to_string(),
+            data.len().to_string(),
+            *fin,
+            None,
+        ),
+        Frame::MaxData { maximum_data } => QuicFrame::max_data(maximum_data.to_string()),
+        Frame::MaxStreamData {
+            stream_id,
+            maximum_stream_data,
+        } => QuicFrame::max_stream_data(
+            stream_id.as_u64().to_string(),
+            maximum_stream_data.to_string(),
+        ),
+        Frame::MaxStreams {
+            stream_type,
+            maximum_streams,
+        } => QuicFrame::max_streams(
+            match stream_type {
+                frame::StreamType::BiDi => qlog::StreamType::Bidirectional,
+                frame::StreamType::UniDi => qlog::StreamType::Unidirectional,
+            },
+            maximum_streams.as_u64().to_string(),
+        ),
+        Frame::DataBlocked { data_limit } => QuicFrame::data_blocked(data_limit.to_string()),
+        Frame::StreamDataBlocked {
+            stream_id,
+            stream_data_limit,
+        } => QuicFrame::stream_data_blocked(
+            stream_id.as_u64().to_string(),
+            stream_data_limit.to_string(),
+        ),
+        Frame::StreamsBlocked {
+            stream_type,
+            stream_limit,
+        } => QuicFrame::streams_blocked(
+            match stream_type {
+                frame::StreamType::BiDi => qlog::StreamType::Bidirectional,
+                frame::StreamType::UniDi => qlog::StreamType::Unidirectional,
+            },
+            stream_limit.as_u64().to_string(),
+        ),
+        Frame::NewConnectionId {
+            sequence_number,
+            retire_prior,
+            connection_id,
+            stateless_reset_token,
+        } => QuicFrame::new_connection_id(
+            sequence_number.to_string(),
+            retire_prior.to_string(),
+            connection_id.len() as u64,
+            hex(&connection_id),
+            hex(stateless_reset_token),
+        ),
+        Frame::RetireConnectionId { sequence_number } => {
+            QuicFrame::retire_connection_id(sequence_number.to_string())
+        }
+        Frame::PathChallenge { data } => QuicFrame::path_challenge(Some(hex(data))),
+        Frame::PathResponse { data } => QuicFrame::path_response(Some(hex(data))),
+        Frame::ConnectionClose {
+            error_code,
+            frame_type,
+            reason_phrase,
+        } => QuicFrame::connection_close(
+            match error_code {
+                frame::CloseError::Transport(_) => qlog::ErrorSpace::TransportError,
+                frame::CloseError::Application(_) => qlog::ErrorSpace::ApplicationError,
+            },
+            error_code.code(),
+            0,
+            String::from_utf8_lossy(&reason_phrase).to_string(),
+            Some(frame_type.to_string()),
+        ),
+        Frame::HandshakeDone => QuicFrame::unknown(0x1e),
+    }
+}
+
+fn pkt_type_to_qlog_pkt_type(ptype: PacketType) -> qlog::PacketType {
```

Comment:
This name smells bad.

Reply:
This is what happens when there are two modules with enums called `PacketType`...

Reply:
to_qlog_packet_type() ?

Or `impl Into<qlog::PacketType> for neqo_transport::PacketType {...}`.

(I realize the limitations of the type system, but 5 underscores seemed excessive.)

Reply:
done, thanks for the suggestions.